include/circt/Dialect/FIRRTL/FIRRTLAttributes.h

```diff
@@ -14,6 +14,110 @@
 #define CIRCT_DIALECT_FIRRTL_FIRRTLATTRIBUTES_H
 
 #include "circt/Dialect/FIRRTL/FIRRTLDialect.h"
+#include "circt/Dialect/FIRRTL/FIRRTLTypes.h"
+#include "circt/Support/LLVM.h"
+#include "llvm/ADT/BitVector.h"
+#include "llvm/ADT/iterator.h"
+
+namespace circt {
+namespace firrtl {
+
+//===----------------------------------------------------------------------===//
+// PortDirections
+//===----------------------------------------------------------------------===//
+
+/// This represents the direction of a single port.
+enum class Direction { Input = 0, Output };
```

Comment:
Nit: it may be better to use `In` and `Out` as this aligns with what the `"in"` and `"out"` we chose for serialized MLIR:
```suggestion
/// This represents the direction of a single port.
enum class Direction { In = 0, Out };
```

Reply:
Also, any reason to do `In = 0` here? I always want to do this, but I it's unnecessary due to the C++ spec, yes?

Reply:
Switched to `In` and `Out`, and got rid of the `= 0`.  @darthscsi  Was the `= 0` there for a reason, e.g. to indicate that the enum order mattered?